shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -565,8 +561,8 @@ - (void)viewDidLoad {
     _engineNeedsLaunch = NO;
   }
 
-  FML_DCHECK([_engine.get() viewController] != nil)
-      << "FlutterViewController::viewWillAppear:AttachView ViewController was nil";
+  FML_DCHECK([_engine.get() viewController] == self)
+      << "FlutterViewController shown but is not attached to a FlutterEngine";
```

Comment:
Won't this message be confusing if `[_engine.get() viewController]` is nil?

Reply:
what scenario do you have in mind? A viewDidLoad on a FlutterViewController should never be called if [_engine.get() viewController] is nil right?

Reply:
Well, `viewDidLoad`  doesn't necessarily  mean that the  ViewController has been  shown.  It just mean's it's  been lazy loaded  (hopefully only when it  gets  shown, but that wasn't the case a couple of months  ago).

Also, what  about the  case where `[_engine.get() viewController] == self && [_engine.get() viewController] != nil`?   The wording is a bit odd in that case.

Reply:
Tweaked a few words from VC being shown to VC's view being loaded. But the spirit of the warning is still the same.

A few months ago, the bug was that the engine loaded the view on the VC earlier than when the VC would have been shown. If an engine wasn't attached altogether, the warning would be even more important. 

> Also, what about the case where [_engine.get() viewController] == self && [_engine.get() viewController] != nil? The wording is a bit odd in that case.

You meant `[_engine.get() viewController] != self && [_engine.get() viewController] != nil` right? That's still a bug (that somehow this VC's view is loaded and isn't attached to an engine).